stablehlo/dialect/StablehloOps.cpp

```diff
@@ -547,12 +547,12 @@ void CustomCallOp::getEffects(
 // CholeskyOp
 //===----------------------------------------------------------------------===//
 
-LogicalResult CholeskyOp::inferReturnTypeComponents(
-    MLIRContext*, Optional<Location> location, ValueShapeRange operands,
+LogicalResult CholeskyOp::inferReturnTypes(
```

Comment:
Same question as in #885. What is the reason for refactoring to `inferReturnTypes`? I think that `inferReturnTypeComponents` recently added support for encodings as well.

Reply:
Yes, you're correct. I rewrote this with the assumption that the support wasn't added yet. I've reverted the changes.